config/routes.rb

```diff
@@ -42,4 +42,9 @@
   # Adding the routes for deleting posts and comments
 delete "/posts/:id", to: "posts#destroy", as: :delete_post
 delete "/comments/:id", to: "comments#destroy", as: :delete_comment
+namespace :api do
+  resources :posts, only: [:index] do
+    resources :comments, only: [:index, :create]
+  end
+end
```

Comment:
I believe that for us to retrieve posts  for a user, we need to wrap the posts resource with users. The reason is because as per your posts controller index method `params[:author_id]` will not have a value

- [ ] Kindly work on this